metakb/transform/civic.py

```diff
@@ -0,0 +1,499 @@
+            elif clinical_significance == 'Resistance':
+                predicate = schemas.PredictivePredicate.RESISTANCE.value
+            elif clinical_significance == 'Reduced Sensitivity':
+                predicate = \
+                    schemas.PredictivePredicate.REDUCED_SENSITIVITY.value
+            elif clinical_significance == 'Adverse Response':
+                predicate = schemas.PredictivePredicate.ADVERSE_RESPONSE.value
+        # TODO
+        elif proposition_type == schemas.PropositionType.DIAGNOSTIC.value:
+            pass
+        elif proposition_type == schemas.PropositionType.PROGNOSTIC.value:
+            pass
+        elif proposition_type == schemas.PropositionType.PREDISPOSING.value:
+            pass
+        elif proposition_type == schemas.PropositionType.FUNCTIONAL.value:
+            pass
+        elif proposition_type == schemas.PropositionType.ONCOGENIC.value:
+            pass
+        return predicate
+
+    def _get_variation_descriptors(self, variant, gene):
+        """Return a list of Variation Descriptors.
+
+        :param dict variant: A CIViC variant record
+        :param dict gene: A CIViC gene record
+        :return: A list of Variation Descriptors
+        """
+        structural_type = None
+        molecule_context = None
+        if len(variant['variant_types']) == 1:
+            # TODO: Go through SO to find the molecule_context
+            #  Is there a better way to do this?
+            so_id = variant['variant_types'][0]['so_id']
+            if so_id in ['SO:0001583', 'SO:0001818']:
+                structural_type = 'SO:0001060'
+                molecule_context = 'protein'
+            elif so_id in ['SO:0001886', 'SO:0001576', 'SO:0001889']:
+                structural_type = 'SO:0001060'
+                molecule_context = 'transcript'
+            else:
+                # TODO: Genomic
+                pass
+
+        variant_query = f"{gene['name']} {variant['name']}"
+
+        try:
+            validations = self.variant_to_vrs.get_validations(variant_query)
+        except:  # noqa: E722
+            logger.error(f"toVRS: {variant_query}")
+            return []
+        normalized_resp = \
+            self.variant_normalizer.normalize(variant_query,
+                                              validations,
+                                              self.amino_acid_cache)
+
+        if not normalized_resp:
+            # TODO: Maybe we can search on the hgvs expression??
+            logger.warning(f"{variant_query} is not yet supported in"
+                           f" Variant Normalization normalize.")
+            return []
+
+        variation_descriptor = schemas.VariationDescriptor(
+            id=f"civic:vid{variant['id']}",
+            label=variant['name'],
+            description=variant['description'],
+            value_id=normalized_resp.value_id,
+            value=normalized_resp.value,
+            gene_context=f"civic:gid{gene['id']}",
```

Comment:
`gene_context` is probably wrong. I think we'd put the `coordinates.representative_transcript` (if this value exists) from a [variant record](https://civicdb.org/api/variants/38), i.e. `gene_context="ensembl:ENST00000269571.5"`?